packages/gafl-webapp-service/src/__tests__/server.spec.js

```diff
@@ -40,23 +40,25 @@ describe('The server', () => {
     )
   })
 
-  describe('handles process interrupts', () => {
-    it.each(['SIGINT', 'SIGTERM'])('implements a shutdown handler to respond to the %s signal', (signal, done) => {
+  describe.each(['SIGINT', 'SIGTERM'])('implements a shutdown handler to respond to the %s signal', signal => {
+    it('calls the hapi shutdown hook', done => {
       jest.isolateModules(async () => {
         const { shutdownBehavior, createServer, init } = require('../server.js')
         try {
           createServer((({ port, ...l }) => l)(catboxOptions))
           await init()
           shutdownBehavior()
-          jest.spyOn(process, 'exit').mockImplementation(async () => {
-            expect(true).toBeTruthy()
-            // Allows time for the server to shut down
-            await new Promise(resolve => setTimeout(resolve, 1000))
-            done()
-          })
-          process.emit(signal)
+          const serverStopSpy = jest.spyOn(server, 'stop').mockImplementation(jest.fn())
```

Comment:
This is odd, because I could not get this very code to work. It kept giving me timeout errors in the jests tests until I eliminated it. It ran ok on its own, but when running the whole suite it was failing, repeatedly. Keep an eye on this.

Reply:
I think the timeout errors were occurring because the function wasn't previously calling `await server.stop()` after  `jest.restoreAllMocks()` isn't called, which I believe was also why I was seeing Jest forcibly terminate the tests in the frontend.